src/mono/mono/mini/mini-amd64.c

```diff
@@ -7600,13 +7600,17 @@ mono_arch_output_basic_block (MonoCompile *cfg, MonoBasicBlock *bb)
 					amd64_mov_membase_reg (code, ins->sreg1, MONO_STRUCT_OFFSET (MonoContext, gregs) + i * sizeof (target_mgreg_t), i, sizeof (target_mgreg_t));
 			break;
 		default:
+			#ifndef DISABLE_LOGGING
 			g_warning ("unknown opcode %s in %s()\n", mono_inst_name (ins->opcode), __FUNCTION__);
+			#endif
 			g_assert_not_reached ();
 		}
-
+		
+		#ifndef DISABLE_LOGGING
 		g_assertf ((code - cfg->native_code - offset) <= max_len,
 			   "wrong maximal instruction length of instruction %s (expected %d, got %d)",
 			   mono_inst_name (ins->opcode), max_len, (int)(code - cfg->native_code - offset));
+		#endif
```

Comment:
this is removing the `g_assertf` completely so we'd be missing the check, which is not what we want.